.github/workflows/main.yml

```diff
@@ -4,27 +4,50 @@ on: [push, pull_request]
 
 jobs:
   Build:
-    name: ${{ matrix.platform.name }}
-    runs-on: ${{ matrix.platform.os }}
+    name: '${{ matrix.platform.name }}'
+    runs-on: '${{ matrix.platform.os }}'
     strategy:
       fail-fast: false
       matrix:
         platform:
-        - { name: Windows, os: windows-latest, msvc: true }
-        - { name: Linux,   os: ubuntu-latest }
-        - { name: MacOS,   os: macos-latest }
+        - { name: 'Linux',                    os: ubuntu-latest,  shell: sh }
+        - { name: 'MacOS',                    os: macos-latest,   shell: sh }
+        - { name: 'Windows msys2 (mingw32)',  os: windows-latest, shell: 'msys2 {0}', msystem: mingw32, msys-env: mingw-w64-i686 }
+        - { name: 'Windows msys2 (mingw64)',  os: windows-latest, shell: 'msys2 {0}', msystem: mingw64, msys-env: mingw-w64-x86_64 }
+        - { name: 'Windows msys2 (clang32)',  os: windows-latest, shell: 'msys2 {0}', msystem: clang32, msys-env: mingw-w64-clang-i686 }
+        - { name: 'Windows msys2 (clang64)',  os: windows-latest, shell: 'msys2 {0}', msystem: clang64, msys-env: mingw-w64-clang-x86_64 }
```

Comment:
I don't understand why clang64 is failing during CMake configuration.
It fails in libsdl-org/setup-sdl, but also [during normal CMake configuration](https://github.com/madebr/sdl2-compat/actions/runs/5777560250/job/15657876918).

Reply:
Don't know about that one..

The clang32 one does configure, but fails linkage because it can't do `-lgcc`: maybe adjust stuff so it links `compiler_rt` or whatever the clang version is supposed to be?

Reply:
Looks like I couldn't do `path-type: inherit` for the `msys2/setup-msys2` action, when using clang64.